File: packages/react-native-nitro-fetch/ios/NitroLogger.swift
import Foundation

/// Flip `enabled` to log in release builds too.
enum NitroLogger {
#if DEBUG
  static var enabled = true
#else
  static var enabled = false
#endif

  static func log(_ message: @autoclosure () -> String) {
    if enabled { print(message()) }
  }
}

File: packages/react-native-nitro-fetch/android/src/main/java/com/margelo/nitro/nitrofetch/NitroLogger.kt
package com.margelo.nitro.nitrofetch

import android.util.Log

/** Flip [enabled] to log in release builds too. */
object NitroLogger {
  @JvmField var enabled: Boolean = BuildConfig.DEBUG

  fun d(tag: String, msg: String) { if (enabled) Log.d(tag, msg) }
  fun i(tag: String, msg: String) { if (enabled) Log.i(tag, msg) }
  fun w(tag: String, msg: String, t: Throwable? = null) { if (enabled) Log.w(tag, msg, t) }
  fun e(tag: String, msg: String, t: Throwable? = null) { if (enabled) Log.e(tag, msg, t) }
}
